### PR TITLE
Parameters with study import from existing case

### PR DESCRIPTION
**Pull request by archendel**

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -124,12 +124,23 @@ public ResponseEntity<List<CreatedStudyBasicInfos>> getStudyListMetadata(@Reques
...
+        BasicStudyInfos createStudy = studyService.createStudy(caseUuid, userId, studyUuid, importParameters);
         return ResponseEntity.ok().body(createStudy);
     }
 
+    @GetMapping(value = "/studies/cases/{caseUuid}/import-parameters")
+    @Operation(summary = "get import parameters for specific case")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "The import formats of the case"),
+        @ApiResponse(responseCode = "404", description = "The case can not be found")})
+    public ResponseEntity<String> getCaseImportFormats(@PathVariable("caseUuid") UUID caseUuid) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:139]: Perhaps it's better to use DTO `ImportExportFormatMeta` ?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 2]
```diff
@@ -285,6 +285,20 @@ public String getCaseName(UUID studyUuid) {
         }
     }
 
+    private String getFormatImportParameters(String format) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_CONVERSION_API_VERSION + "/import/formats/{format}/parameters")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyService.java:289]: Perhaps it's better to use DTO ImportExportFormatMeta ?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 3]
```diff
@@ -625,6 +639,14 @@ public String getCaseFormat(UUID caseUuid, UUID studyUuid, String userId) {
         }
     }
 
+    public String getCaseFormat(UUID caseUuid) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyService.java:642]: Method already exists ?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 3 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyService.java:642]: method exists with a different signature, I'm not creating a study here
maybe we should refactor "getCaseFormat" original method ?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 4]
```diff
@@ -1767,6 +1745,14 @@ public Consumer<Message<String>> consumeSaFailed() {
         };
     }
 
+//    @Bean
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyService.java:1748]: ??

Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyService.java:1748]: removed

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**